### PR TITLE
doc improvement: fix broken links, minor fixes, and toc update

### PR DESCRIPTION
**Pull request by divipillai**
* Fixed broken links and toc update.
* Fixed a minor update to the nRF desktop application.
* Arranged Applications in alphabetical order.
* Update with respect to nRF9151.

### CODE REVIEW

**Review by tomi-font — APPROVED** [review]
Approving SLM changes.

**Review by wiba-nordic — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

File: doc/nrf/applications.rst [hunk 1]
```diff
@@ -17,13 +17,13 @@ If you want to list applications available for one or more specific boards, `use
...
-   ../../applications/asset_tracker_v2/README
-   ../../applications/serial_lte_modem/README
-   ../../applications/nrf_desktop/README
    ../../applications/nrf5340_audio/index
+   ../../applications/nrf_desktop/README
    ../../applications/machine_learning/README
    ../../applications/matter_weather_station/README
    ../../applications/zigbee_weather_station/README
-   ../../applications/ipc_radio/README
+   ../../applications/serial_lte_modem/README
```
Comment (peknis) [inline doc/nrf/applications.rst:29]: Serial modem before Thingies.

File: doc/nrf/releases_and_maturity/known_issues.rst [hunk 2]
```diff
@@ -4333,7 +4333,7 @@ KRKNWK-13063: RTT logs do not work with the Light Bulb multiprotocol sample with
 KRKNWK-13064: Nordic DFU is not compliant with HAP certification requirements
   Some of the HAP certification requirements are not met by the Nordic DFU solution.
 
-  **Workaround:** Cherry-pick changes from `PR #332 in sdk-homekit repo`_.
+  **Workaround:** Cherry-pick changes from PR #332 in sdk-homekit repo.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/known_issues.rst:4336]:
```suggestion
  **Workaround:** Cherry-pick changes from PR #332 in the sdk-homekit repository.
```

File: doc/nrf/releases_and_maturity/known_issues.rst [hunk 3]
```diff
@@ -4351,7 +4351,7 @@ KRKNWK-13095: Change in KVS key naming scheme causes an error for updated device
   A previous implementation allowed for empty key in domain.
   This has been restricted during refactoring.
 
-  **Workaround:** Cherry-pick changes from `PR #329 in sdk-homekit repo`_.
+  **Workaround:** Cherry-pick changes from PR #329 in sdk-homekit repo.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/known_issues.rst:4354]:
```suggestion
  **Workaround:** Cherry-pick changes from PR #329 in the sdk-homekit repository.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Comment by divipillai** [comment]
@carlescufi , @gmarull : Some of the links on this table are broken.  
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/repository_revisions.html#ncs-repository-revisions. The table is pulled from a script. The behavior is seen in the previous releases also. For example, qcbor -
[751d36583a9ce1a640900c57e13c9b6b8f3a2ba2](https://github.com/laurencelundblade/QCBOR.git/commit/751d36583a9ce1a640900c57e13c9b6b8f3a2ba2), dragoon etc.

Could you confirm how this can be fixed.

**Review by MarekPieta — APPROVED** [review]
nRF Desktop changes look good. Thanks for the fixes

**Review by jayteemo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Comment by divipillai** [comment]
Also, added files to fix broken links in the Matter repo ( provided by @wiba-nordic).

**Review by juhaylinen — APPROVED** [review]

**Comment by jfischer-no** [comment]
@divipillai Do not update the revision yet, wait until https://github.com/nrfconnect/sdk-nrf/pull/14282 get merged.

**Comment by jfischer-no** [comment]
> @divipillai Do not update the revision yet, wait until #14282 get merged.

@divipillai Please update.

**Comment by divipillai** [comment]
> > @divipillai Do not update the revision yet, wait until #14282 get merged.
> 
> @divipillai Please update.

Updated.

**Pull request merged**